src/Shared/NetCore.cs

```diff
@@ -1,15 +1,16 @@
-﻿using System;
-using System.Reflection;
-
-public class NetCore
-{
-    public static string GetNetCoreVersion()
-    {
-        var assembly = typeof(System.Runtime.GCSettings).GetTypeInfo().Assembly;
-        var assemblyPath = assembly.CodeBase.Split(new[] { '/', '\\' }, StringSplitOptions.RemoveEmptyEntries);
-        int netCoreAppIndex = Array.IndexOf(assemblyPath, "Microsoft.NETCore.App");
-        if (netCoreAppIndex > 0 && netCoreAppIndex < assemblyPath.Length - 2)
-            return assemblyPath[netCoreAppIndex + 1];
-        throw new Exception("Unable to determine .NET Core version.");
-    }
-}
+﻿using System;
+using System.Reflection;
+
+public class NetCore
+{
+    public static string GetNetCoreVersion()
+    {
+        Assembly assembly = typeof(System.Runtime.GCSettings).GetTypeInfo().Assembly;
+        var assemblyCodeBase = assembly.CodeBase ?? throw new ArgumentNullException("assemblyCodeBase");
```

Comment:
I don't think this is the correct exception type since "assemblyCodeBase" is not an argument. I would suggest using an `InvalidOperationException`.